bin/bootstrap_practice_exercise.exs

```diff
@@ -0,0 +1,258 @@
+    end)}) :: {:ok, String.t()} | {:error, String.t()}
+     def #{property}(#{Enum.map_join(variables, ", ", fn {var, _} -> var end)}) do
+     end
+    """
+  end
+
+  def print_comments(comments, do_not_print) do
+    comments
+    |> Enum.reject(fn {field, _value} -> field in do_not_print end)
+    |> Enum.map_join("\n", fn
+      {"comments", values} when is_list(values) ->
+        "# #{Enum.map_join(values, "\n# ", &String.trim/1)}"
+
+      {field, values} when is_list(values) ->
+        "#\n# --#{field} --\n# #{Enum.map_join(values, "\n# ", &inspect/1)}"
+
+      {field, value} ->
+        "#\n# -- #{field} --\n# #{inspect(value)}"
+    end)
+  end
+
+  def print_input(%{} = input) do
+    Enum.map_join(input, "\n", fn {variable, value} -> "#{variable} = #{inspect(value)}" end)
+  end
+
+  def print_input(input), do: "input = #{inspect(input)}"
+
+  def print_variables(%{} = input),
+    do: Enum.map_join(input, ", ", fn {variable, _value} -> variable end)
+
+  def print_variables(input), do: input
+
+  def print_expected(%{"error" => err}), do: "{:error, #{inspect(err)}}"
+  def print_expected(expected), do: "{:ok, #{inspect(expected)}}"
+
+  def print_test_case(%{"description" => description, "cases" => sub_cases} = c, module) do
+    """
+    describe \"#{description}\" do
+    #{Generate.print_comments(c, ["description", "cases"])}
+    #{Enum.map_join(sub_cases, "\n\n", &Generate.print_test_case(&1, module))}
+    end
+    """
+  end
+
+  def print_test_case(
+        %{
+          "description" => description,
+          "property" => property,
+          "input" => input,
+          "expected" => expected
+        } = c,
+        module
+      ) do
+    """
+    @tag :pending
+    test \"#{description}\" do
+      #{Generate.print_comments(c, ["description", "property", "input", "expected", "uuid"])}
+      #{print_input(input)}
+      output = #{module}.#{Generate.to_snake_case(property)}(#{print_variables(input)})
+      expected = #{print_expected(expected)}
+
+      assert output == expected
+    end
+    """
+  end
+end
+
+Mix.install([
+  {:jason, "~> 1.2"},
+  {:toml, "~> 0.6"}
+])
+
+[exercise] = System.argv()
+
+exercise_path = String.replace(exercise, "-", "_")
+
+module =
+  exercise
+  |> String.split("-")
+  |> Enum.map_join("", &String.capitalize/1)
+
+## Step 1: create folder structure
+
+Mix.Generator.create_directory("exercises/practice/#{exercise}")
+Mix.Generator.create_directory("exercises/practice/#{exercise}/.docs")
+Mix.Generator.create_directory("exercises/practice/#{exercise}/.meta")
+Mix.Generator.create_directory("exercises/practice/#{exercise}/lib")
+Mix.Generator.create_directory("exercises/practice/#{exercise}/test")
+
+## Step 2: add common files
+
+# .formatter.exs
+format = """
+# Used by "mix format"
+[
+  inputs: ["{mix,.formatter}.exs", "{config,lib,test}/**/*.{ex,exs}"]
+]
+"""
+
+Mix.Generator.create_file("exercises/practice/#{exercise}/.formatter.exs", format)
```

Comment:
I had no idea about this module 😍 awesome

Reply:
It is quite nice. And actually it wouldn't be much more work to make this script into a `Mix.Task`. It would be kind of cool: `mix ex.practice.gen darts`.